crml/nft/src/benchmarking.rs

```diff
@@ -0,0 +1,306 @@
+	}
+
+	burn {
+		let owner: T::AccountId = account("owner", 0, 0);
+		let collection_id = setup_token::<T>(owner.clone());
+
+		let token_id = (collection_id, 0, 0);
+	}: _(RawOrigin::Signed(owner.clone()), token_id)
+	verify {
+		assert_eq!(<Nft<T>>::token_owner((collection_id, 0), 0), T::AccountId::default());
+	}
+
+	sell {
+		let owner: T::AccountId = account("owner", 0, 0);
+		let collection_id = setup_token::<T>(owner.clone());
+		let token_id = (collection_id, 0, 0);
+		let listing_id = <Nft<T>>::next_listing_id();
+
+	}: _(RawOrigin::Signed(owner.clone()), token_id, Some(owner.clone()), PAYMENT_ASSET, PRICE, Some(T::BlockNumber::from(1_u32)))
+	verify {
+		assert!(<Nft<T>>::listings(listing_id).is_some());
+	}
+
+	buy {
+		let owner: T::AccountId = account("owner", 0, 0);
+		let buyer: T::AccountId = account("buyer", 0, 0);
+		let collection_id = setup_token::<T>(owner.clone());
+		let token_id = (collection_id, 0, 0);
+		let listing_id = <Nft<T>>::next_listing_id();
+		let _ = T::MultiCurrency::deposit_creating(&buyer, Some(PAYMENT_ASSET), PRICE);
+		let listing_id = <Nft<T>>::next_listing_id();
+		let _ = <Nft<T>>::sell(RawOrigin::Signed(owner.clone()).into(), token_id, Some(buyer.clone()), PAYMENT_ASSET, PRICE, None).expect("listed ok");
+
+	}: _(RawOrigin::Signed(buyer.clone()), listing_id)
+	verify {
+		assert_eq!(<Nft<T>>::token_owner((collection_id, 0), 0), buyer);
+	}
+
+	bid {
+		let owner: T::AccountId = account("owner", 0, 0);
+		let buyer: T::AccountId = account("buyer", 0, 0);
+		let collection_id = setup_token::<T>(owner.clone());
+		let token_id = (collection_id, 0, 0);
+		let listing_id = <Nft<T>>::next_listing_id();
+		let duration = T::BlockNumber::from(100_u32);
+
+		let _ = T::MultiCurrency::deposit_creating(&owner, Some(PAYMENT_ASSET), PRICE);
+		let _ = T::MultiCurrency::deposit_creating(&buyer, Some(PAYMENT_ASSET), PRICE + 1);
+		let listing_id = <Nft<T>>::next_listing_id();
+		let _ = <Nft<T>>::auction(RawOrigin::Signed(owner.clone()).into(), token_id, PAYMENT_ASSET, PRICE, Some(duration)).expect("listed ok");
+		// worst case path is to replace an existing bid
+		let _ = <Nft<T>>::bid(RawOrigin::Signed(owner.clone()).into(), listing_id, PRICE);
+
+	}: _(RawOrigin::Signed(buyer.clone()), listing_id, PRICE + 1)
+	verify {
+		assert_eq!(<Nft<T>>::listing_winning_bid(listing_id), Some((buyer.clone(), PRICE + 1)));
+	}
+
+	cancel_sale {
+		let owner: T::AccountId = account("owner", 0, 0);
+		let collection_id = setup_token::<T>(owner.clone());
+		let token_id = (collection_id, 0, 0);
+		let listing_id = <Nft<T>>::next_listing_id();
+		let duration = T::BlockNumber::from(100_u32);
+		let listing_id = <Nft<T>>::next_listing_id();
+		let _ = <Nft<T>>::auction(RawOrigin::Signed(owner.clone()).into(), token_id, PAYMENT_ASSET, PRICE, Some(duration)).expect("listed ok");
+
+	}: _(RawOrigin::Signed(owner.clone()), listing_id)
+	verify {
+		assert!(<Nft<T>>::listings(listing_id).is_none());
+	}
+}
+
+#[cfg(test)]
```

Comment:
I think there's a new macro to generate all these tests auto-magically (not sure if we got it in 3.0.0-rc5)

Reply:
Let's keep it the same as all the other tests in the current version and upgrade the tests as a separate step.